src/projects.h

```diff
@@ -327,6 +327,7 @@ struct PJconsts {
     int  geoc;                      /* Geocentric latitude flag */
     int  is_latlong;                /* proj=latlong ... not really a projection at all */
     int  is_geocent;                /* proj=geocent ... not really a projection at all */
+    int  need_ellps;                /* 0 for operations that are purely cartesian */
```

Comment:
Does `need_ellps` also imply that the output has to be scaled by the semimajor axis?

Reply:
No - and it doesn't even mean that it is evident from the outside that an ellipsoid is needed:

Consider e.g. a hypothetical  case where you do a correction of cartesian coordinates.

Input is cartesian, output is cartesian, so no need for ellipsoid is expected.

But the correction grid is in geographical coordinates, so you need the ellipsoid to convert to geographical, in order to fetch the correction to apply to the cartesian.

`need_ellps`  is really just a flag to allow the plumbing  in `pj_init.c` to relax the initial check for whether a spheroid is specified.

Reply:
Good, just wanted to check that we had the same thing in mind. Your case is not very hypothetical since that is exactly what the `deformation` transformation does :)

> In general, I would like to get rid of the scaled output from projections, since i complicates lots of other stuff

Would that also be the case if there was information on the type of operation attached to each `PJ`? That would enable you to skip the scaling logic when not processing a projection.